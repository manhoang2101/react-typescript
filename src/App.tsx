import MasterContainer from "./containers/master";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import ECommonAction from "./store/common/common.actions";
import { ICommonState } from "./store/common/common.types";
import { IAppState } from "./store/types";
export const mapStateToProps = (state: IAppState): ICommonState => ({
  config: state.commonReducer.config,
  pageLoadding: state.commonReducer.pageLoadding,
});
export const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchConfigAction: () => dispatch({ type: ECommonAction.FETCH_CONFIG }),
});
const App = connect(mapStateToProps, mapDispatchToProps)(MasterContainer);
export default App;
