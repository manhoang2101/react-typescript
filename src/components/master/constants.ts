export const drawerWidth = 240;
export const menuId = "primary-search-account-menu";
export const mobileMenuId = "primary-search-account-menu-mobile";
