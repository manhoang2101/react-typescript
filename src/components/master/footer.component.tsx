import React from "react";
import { WithStyles, withStyles } from "@material-ui/core";
import style from "./style";

export interface PropsAppHeader extends WithStyles<typeof style> {
  SetHeight?: (type: string, height: number) => void;
}
export class FooterComponent extends React.Component<PropsAppHeader> {
  refCallback = (element: HTMLElement | null) => {
    if (element) {
      const bounding = element.getBoundingClientRect();
      this.props.SetHeight &&
        this.props.SetHeight("FooterComponent", bounding.height);
    }
  };
  render() {
    return <div ref={this.refCallback}> </div>;
  }
}
export default withStyles(style)(FooterComponent);
