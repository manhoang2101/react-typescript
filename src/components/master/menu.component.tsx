import React from "react";
import { Menu, MenuItem, IconButton, Badge } from "@material-ui/core";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MailIcon from "@material-ui/icons/Mail";
import NotificationsIcon from "@material-ui/icons/Notifications";
export interface PropsMenu {
  AnchorEl?: HTMLElement | null;
  MobileMoreAnchorEl?: HTMLElement | null;
  HandleMenuClose?: () => void;
  HandleMobileMenuClose?: () => void;
  HandleProfileMenuOpen?: (
    event: React.MouseEvent<HTMLElement, MouseEvent>
  ) => void;
  MenuId: string;
  MobileMenuId: string;
  StoryOpen?: boolean;
}
const MenuComponent = (props: PropsMenu) => {
  const isMenuOpen = Boolean(Boolean(props.AnchorEl) || props.StoryOpen);
  const isMobileMenuOpen = Boolean(props.MobileMoreAnchorEl);
  const renderMenu = (
    <Menu
      anchorEl={props.AnchorEl}
      anchorOrigin={{ vertical: "top", horizontal: "right" }}
      id={props.MenuId}
      keepMounted
      transformOrigin={{ vertical: "top", horizontal: "right" }}
      open={isMenuOpen}
      onClose={props.HandleMenuClose}
    >
      <MenuItem onClick={props.HandleMenuClose}>Profile</MenuItem>
      <MenuItem onClick={props.HandleMenuClose}>My account</MenuItem>
    </Menu>
  );

  const renderMobileMenu = (
    <Menu
      anchorEl={props.MobileMoreAnchorEl}
      anchorOrigin={{ vertical: "top", horizontal: "right" }}
      id={props.MobileMenuId}
      keepMounted
      transformOrigin={{ vertical: "top", horizontal: "right" }}
      open={isMobileMenuOpen}
      onClose={props.HandleMobileMenuClose}
    >
      <MenuItem>
        <IconButton aria-label="show 4 new mails" color="inherit">
          <Badge badgeContent={4} color="secondary">
            <MailIcon />
          </Badge>
        </IconButton>
        <p>Messages</p>
      </MenuItem>
      <MenuItem>
        <IconButton aria-label="show 11 new notifications" color="inherit">
          <Badge badgeContent={11} color="secondary">
            <NotificationsIcon />
          </Badge>
        </IconButton>
        <p>Notifications</p>
      </MenuItem>
      <MenuItem onClick={props.HandleProfileMenuOpen}>
        <IconButton
          aria-label="account of current user"
          aria-controls="primary-search-account-menu"
          aria-haspopup="true"
          color="inherit"
        >
          <AccountCircle />
        </IconButton>
        <p>Profile</p>
      </MenuItem>
    </Menu>
  );
  return (
    <>
      {renderMenu}
      {renderMobileMenu}
    </>
  );
};

export default MenuComponent;
