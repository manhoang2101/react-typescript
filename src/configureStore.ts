import { Store, createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import { routerMiddleware } from "connected-react-router";
import { composeWithDevTools } from "redux-devtools-extension";
import { History } from "history";
import { createRootReducer, rootSaga } from "./store";
import { IAppState } from "./store/types";

export default function configureStore(history: History): Store<IAppState> {
  const composeEnhancers = composeWithDevTools({});
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    createRootReducer(history),
    composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);
  return store;
}
