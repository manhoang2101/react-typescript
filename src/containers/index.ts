import { VariantType } from "notistack";
import { IConfig } from "../stories/common/common.types";
import { Cookies } from "react-cookie";

export default interface ContainerProp {
  openNotification: (
    variant: VariantType,
    massage: string,
    option?: Object,
    callBack?: () => void
  ) => void;
  config?: IConfig;
  classes?: any;
  showFooter?: (show: boolean) => void;
  showHeader?: (show: boolean) => void;
  history?: any;
  cookies: Cookies;
}
