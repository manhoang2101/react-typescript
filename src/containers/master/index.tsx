import HeaderComponent from "../../components/master/header.component";
import React from "react";
import FooterComponent from "../../components/master/footer.component";
import { AppRoutes } from "../../untils/routers";
import { Switch, Route, BrowserRouter as Router } from "react-router-dom";
import styles from "../../untils/styles";
import {
  WithStyles,
  withStyles,
  Backdrop,
  CircularProgress,
} from "@material-ui/core";
import clsx from "clsx";
import { SnackbarProvider, VariantType, useSnackbar } from "notistack";
import { IConfig } from "../../stories/common/common.types";
export interface PropsMaster extends WithStyles<typeof styles> {
  fetchConfigAction: () => void;
  openNotification: (callBack: () => void) => void;
  config: IConfig;
  pageLoading: boolean;
}
export interface StatesMaster {
  OpenSidebar: boolean;
  HeightHeader: number;
  HeightFooter: number;
}
export class MasterContainer extends React.Component<
  PropsMaster,
  StatesMaster
> {
  constructor(props: Readonly<PropsMaster>) {
    super(props);
    this.state = {
      OpenSidebar: false,
      HeightHeader: 0,
      HeightFooter: 0,
    };
    this.handleOpenSidebar.bind(this);
    this.setHeightElement.bind(this);
    this.props.fetchConfigAction();
  }
  handleOpenSidebar = (status: boolean) => {
    this.setState({
      OpenSidebar: status,
    });
  };
  handOpenNotification = (variant: VariantType, massage: string) => {
    const { openNotification } = this.props;
    const { enqueueSnackbar } = useSnackbar();
    openNotification(() => {
      enqueueSnackbar(massage, { variant });
    });
  };
  setHeightElement = (name: string, height: number) => {
    switch (name) {
      case "HeaderComponent":
        this.setState({
          HeightHeader: height,
        });
        break;
      case "FooterComponent":
        this.setState({
          HeightFooter: height,
        });
        break;
    }
  };
  render() {
    const { classes } = this.props;
    return (
      <SnackbarProvider maxSnack={3}>
        <HeaderComponent
          Open={this.state.OpenSidebar}
          SetOpen={this.handleOpenSidebar}
          SetHeight={this.setHeightElement}
          key="HeaderComponent"
        />
        <main
          style={{
            marginTop: this.state.HeightHeader,
            marginBottom: this.state.HeightFooter,
          }}
          className={clsx(classes.content, {
            [classes.contentShift]: this.state.OpenSidebar,
          })}
        >
          <Router>
            <Switch key="Switch">
              {AppRoutes.map((route) => (
                <Route
                  exact
                  path={route.path}
                  key={route.label}
                  pathMatch={route?.pathMatch}
                >
                  {
                    <route.component
                      openNotification={this.handOpenNotification}
                    />
                  }
                </Route>
              ))}
            </Switch>
          </Router>
          <Backdrop className={classes.backdrop} open={this.props.pageLoading}>
            <CircularProgress color="inherit" />
          </Backdrop>
        </main>
        <FooterComponent
          SetHeight={this.setHeightElement}
          key="FooterComponent"
        />
      </SnackbarProvider>
    );
  }
}

export default withStyles(styles)(MasterContainer);
