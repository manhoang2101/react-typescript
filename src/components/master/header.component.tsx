import React from "react";
import { WithStyles, withStyles } from "@material-ui/core/styles";
import clsx from "clsx";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import Badge from "@material-ui/core/Badge";
import MenuIcon from "@material-ui/icons/Menu";
import SearchIcon from "@material-ui/icons/Search";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MailIcon from "@material-ui/icons/Mail";
import NotificationsIcon from "@material-ui/icons/Notifications";
import MoreIcon from "@material-ui/icons/MoreVert";
import style from "./style";
import MenuComponent from "./menu.component";
import SidebarComponent from "./sidebar.component";
import { menuId, mobileMenuId } from "./constants";
export interface IAppHeaderProps extends WithStyles<typeof style> {
  Open: boolean;
  SetOpen?: (type: boolean) => void;
  SetHeight?: (type: string, height: number) => void;
}
export interface IAppHeaderStates {
  anchorEl: null | HTMLElement;
  mobileMoreAnchorEl: null | HTMLElement;
}

export class HeaderComponent extends React.Component<
  IAppHeaderProps,
  IAppHeaderStates
> {
  containerRef: any;
  constructor(props: Readonly<IAppHeaderProps>) {
    super(props);
    this.state = {
      anchorEl: null,
      mobileMoreAnchorEl: null,
    };
    this.containerRef = React.createRef();
  }
  refCallback = (element: HTMLElement) => {
    if (element) {
      const bounding = element.getBoundingClientRect();
      this.props.SetHeight &&
        this.props.SetHeight("HeaderComponent", bounding.height);
    }
  };
  componentDidMount() {}
  handleProfileMenuOpen = (event: React.MouseEvent<HTMLElement>): void => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleMobileMenuClose = () => {
    this.setState({ mobileMoreAnchorEl: null });
  };

  handleMenuClose = () => {
    this.setState({ anchorEl: null });
    this.handleMobileMenuClose();
  };

  handleMobileMenuOpen = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
    this.setState({ mobileMoreAnchorEl: event.currentTarget });
  };

  handleDrawerOpen = () => {
    this.props.SetOpen && this.props.SetOpen(true);
  };

  handleDrawerClose = () => {
    this.props.SetOpen && this.props.SetOpen(false);
  };
  render() {
    const { classes } = this.props;
    return (
      <div className={classes.grow}>
        <AppBar
          ref={this.refCallback}
          position="fixed"
          className={clsx(classes.appBar, {
            [classes.appBarShift]: this.props.Open,
          })}
        >
          <Toolbar>
            <IconButton
              edge="start"
              onClick={this.handleDrawerOpen}
              color="inherit"
              aria-label="open drawer"
              className={`${clsx(
                classes.menuButton,
                this.props.Open && classes.hide
              )} button-control-sidebar`}
            >
              <MenuIcon />
            </IconButton>
            <Typography className={classes.title} variant="h6" noWrap>
              3SHU-UI
            </Typography>
            <div className={classes.search}>
              <div className={classes.searchIcon}>
                <SearchIcon />
              </div>
              <InputBase
                placeholder="Searchâ€¦"
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput,
                }}
                inputProps={{ "aria-label": "search" }}
              />
            </div>
            <div className={classes.grow} />
            <div className={classes.sectionDesktop}>
              <IconButton aria-label="show 4 new mails" color="inherit">
                <Badge badgeContent={4} color="secondary">
                  <MailIcon />
                </Badge>
              </IconButton>
              <IconButton
                aria-label="show 17 new notifications"
                color="inherit"
              >
                <Badge badgeContent={17} color="secondary">
                  <NotificationsIcon />
                </Badge>
              </IconButton>
              <IconButton
                edge="end"
                aria-label="account of current user"
                aria-controls={menuId}
                aria-haspopup="true"
                onClick={this.handleProfileMenuOpen}
                color="inherit"
              >
                <AccountCircle />
              </IconButton>
            </div>
            <div className={classes.sectionMobile}>
              <IconButton
                aria-label="show more"
                aria-controls={mobileMenuId}
                aria-haspopup="true"
                onClick={this.handleMobileMenuOpen}
                color="inherit"
              >
                <MoreIcon />
              </IconButton>
            </div>
          </Toolbar>
        </AppBar>
        <MenuComponent
          AnchorEl={this.state.anchorEl}
          MobileMoreAnchorEl={this.state.mobileMoreAnchorEl}
          HandleMenuClose={this.handleMenuClose}
          HandleMobileMenuClose={this.handleMobileMenuClose}
          HandleProfileMenuOpen={this.handleProfileMenuOpen}
          MenuId={menuId}
          MobileMenuId={mobileMenuId}
        ></MenuComponent>
        <SidebarComponent
          Open={this.props.Open}
          HandleDrawerClose={this.handleDrawerClose}
        ></SidebarComponent>
      </div>
    );
  }
}

export default withStyles(style)(HeaderComponent);
