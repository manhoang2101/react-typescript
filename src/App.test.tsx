import configureStore from "redux-mock-store";
import React from "react";
import { Provider } from "react-redux";
import { shallow } from "enzyme";
import { render } from "@testing-library/react";
import App from "./App";
import { MasterContainer } from "./containers/master";

const initialState = {
  commonReducer: {
    config: {},
    pageLoadding: false,
  },
  userReducer: {
    users: [],
    user: null,
  },
};
const mockStore = configureStore();
let store: any;
describe("App", () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });
  test("render component", () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
