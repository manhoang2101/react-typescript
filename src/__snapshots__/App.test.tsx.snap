// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render component 1`] = `
<div>
  <div
    class="HeaderComponent-grow-5"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary HeaderComponent-appBar-14 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root HeaderComponent-menuButton-6 button-control-sidebar MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root HeaderComponent-title-7 MuiTypography-h6 MuiTypography-noWrap"
        >
          Material-UI
        </h6>
        <div
          class="HeaderComponent-search-8"
        >
          <div
            class="HeaderComponent-searchIcon-9"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root HeaderComponent-inputRoot-10"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input HeaderComponent-inputInput-11"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="HeaderComponent-grow-5"
        />
        <div
          class="HeaderComponent-sectionDesktop-12"
        >
          <button
            aria-label="show 4 new mails"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <span
                class="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                >
                  4
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="show 17 new notifications"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <span
                class="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                >
                  17
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="primary-search-account-menu"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="HeaderComponent-sectionMobile-13"
        >
          <button
            aria-controls="primary-search-account-menu-mobile"
            aria-haspopup="true"
            aria-label="show more"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked SidebarComponent-drawer-164"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper SidebarComponent-drawerPaper-165 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
      >
        <div
          class="SidebarComponent-drawerHeader-166"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root handle-drawer-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Inbox
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Starred
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Send email
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Drafts
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                All mail
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Trash
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Spam
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
  <main
    class="MasterContainer-content-1"
    style="margin-top: 0px; margin-bottom: 0px;"
  >
    <h1>
      Home
    </h1>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MasterContainer-backdrop-3"
      style="opacity: 0; visibility: hidden;"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </main>
  <div>
     
  </div>
</div>
`;
